### PR TITLE
Update MaterialScrollBehavior.buildScrollbar for horizontal axes

### LINKED ISSUE #87739: MaterialScrollBehavior.buildScrollbar needs update
The updated recommendation is that for horizontal scroll views, there should be an always visible scrollbar. This is because horizontal scrollable content is much less discoverable.

### PR DESCRIPTION
**Pull request by Piinks**
This updates MaterialScrollBehavior.buildScrollbar to match the latest guidance from Material Design. 

> Horizontal scrollbars have a lower discoverability. When using horizontal scrollbars at the bottom of the table, it should always be visible. If possible, avoid using horizontal scrolling. Consider if the need is for paging rather than scrolling.

Fixes https://github.com/flutter/flutter/issues/87739

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/scrollbar_test.dart [hunk 1]
```diff
@@ -1557,4 +1557,20 @@ void main() {
         ),
     );
   });
+
+  testWidgets('MaterialScrollBehavior applies AlwaysShown Scrollbar', (WidgetTester tester) async {
```
Comment (goderbauer) [inline packages/flutter/test/material/scrollbar_test.dart:1561]: ... to horizontal scrollers

**Review by goderbauer — APPROVED** [review]
LGTM, but looks like checks are unhappy

**Comment by Piinks** [comment]
I see, I forgot TabBars can scroll horizontally. I have opted them out as it does not match the spec to put a scrollbar on a tab bar.

**Comment by Piinks** [comment]
Looks like this needs a little more tweaking. I may not be able to set `isAlwaysShown` to true at first. May require a migration.

**Comment by goderbauer** [comment]
Also: Did we already opt things like the Pageview out of scrollbars? That one probably also shouldn't show them...

**Comment by Piinks** [comment]
Yeah, PageView is already opted out

**Pull request merged**